### PR TITLE
Revert usage of `HTTPClient.shared`

### PR DESCRIPTION
**Pull request by MahdiBM**
Revert commits related to using `HTTPClient.shared`.
I don't think using the shared HTTPClient is necessary for Penny. It only moves Penny out of the control of the `HTTPClient`.

### CODE REVIEW

File: Lambdas/AutoFaqs/AutoFaqsHandler.swift [hunk 1]
```diff
@@ -14,8 +16,12 @@ struct AutoFaqsHandler: LambdaHandler {
     let autoFaqsRepo: S3AutoFaqsRepository
 
     init(context: LambdaInitializationContext) async {
-        self.awsClient = AWSClient()
-        self.autoFaqsRepo = S3AutoFaqsRepository(awsClient: awsClient, logger: context.logger)
+        let httpClient = HTTPClient(
```
Comment (0xTim) [inline Lambdas/AutoFaqs/AutoFaqsHandler.swift:19]: Do we need to worry about shutting this down?

Comment (MahdiBM) [reply Lambdas/AutoFaqs/AutoFaqsHandler.swift:19]: I don't think so considering it'll only need to shutdown when the process is killed.

Comment (MahdiBM) [reply Lambdas/AutoFaqs/AutoFaqsHandler.swift:19]: See: https://swift-open-source.slack.com/archives/C5GMX4LP9/p1682667872540799?thread_ts=1682620274.444989&cid=C5GMX4LP9

File: Sources/Penny/Penny.swift [hunk 2]
```diff
@@ -9,12 +10,34 @@ struct Penny {
...
+            eventLoopGroup: MultiThreadedEventLoopGroup.singleton,
+            configuration: .forPenny
+        )
+        let awsClient = AWSClient(httpClient: httpClient)
 
-        let bot = try await mainService.makeBot()
+        /// These shutdown calls are only useful for tests where we call `Penny.main()` repeatedly
+        defer {
+            /// Shutdown in reverse order of dependance.
+            try! awsClient.syncShutdown()
```
Comment (0xTim) [inline Sources/Penny/Penny.swift:24]: We should not be doing this - this is a Swift bug in that it's not showing you a warning but these methods are (or should be) marked with `noasync` - we need to call the async shutdowns, which unfortunately means no defers

**Review by 0xTim — CHANGES_REQUESTED** [review]

File: Sources/Penny/Penny.swift (hunk 2 above)
Comment (MahdiBM) [reply Sources/Penny/Penny.swift]: Removed all `syncShutdown()`, let's see how the tests go.

File: Sources/Penny/Penny.swift [hunk 3]
```diff
@@ -23,5 +39,10 @@ struct Penny {
         for await event in await bot.events {
             EventHandler(event: event, context: context).handle()
         }
+
+        /// These shutdown calls are only useful for tests where we call `Penny.main()` repeatedly
+        /// Shutdown in reverse order of dependance.
+        try! await awsClient.shutdown()
+        try! await httpClient.shutdown()
```
Comment (0xTim) [inline Sources/Penny/Penny.swift:46]: Any reason for force try these?

File: Sources/Penny/Services/AutoFaqsService/DefaultAutoFaqsService.swift [hunk 4]
```diff
@@ -49,7 +49,7 @@ actor DefaultAutoFaqsService: AutoFaqsService {
         }
     }
 
-    let httpClient: HTTPClient = .shared
+    var httpClient: HTTPClient!
```
Comment (0xTim) [inline Sources/Penny/Services/AutoFaqsService/DefaultAutoFaqsService.swift:52]: Why is this an IUO

File: Sources/Penny/Services/FaqsService/DefaultFaqsService.swift [hunk 5]
```diff
@@ -10,7 +10,7 @@ import NIOHTTP1
 
 actor DefaultFaqsService: FaqsService {
 
-    let httpClient: HTTPClient = .shared
+    var httpClient: HTTPClient!
```
Comment (0xTim) [inline Sources/Penny/Services/FaqsService/DefaultFaqsService.swift:13]: IUO?

File: Tests/PennyTests/Tests/LeafRenderTests.swift [hunk 6]
```diff
@@ -10,24 +10,34 @@ import NIOPosix
...
+            on: httpClient.eventLoopGroup.next()
         )
     )
 
     override func setUp() async throws {
         FakeResponseStorage.shared = FakeResponseStorage()
     }
 
+    override func tearDown() async throws {
+        try! await httpClient.shutdown()
```
Comment (0xTim) [inline Tests/PennyTests/Tests/LeafRenderTests.swift:38]:
```suggestion
        try await httpClient.shutdown()
```

**Review by 0xTim — APPROVED** [review]
A few comments but nothing blocking

File: Sources/Penny/Penny.swift (hunk 3 above)
Comment (MahdiBM) [reply Sources/Penny/Penny.swift:46]: not really, just copy-pasta.

File: Sources/Penny/Services/FaqsService/DefaultFaqsService.swift (hunk 5 above)
Comment (MahdiBM) [reply Sources/Penny/Services/FaqsService/DefaultFaqsService.swift:13]: There is a reason although i don't remember it off-hand.

Comment (MahdiBM) [reply Sources/Penny/Services/FaqsService/DefaultFaqsService.swift:13]: They are likely initialized first, then populated with the http client.

**Pull request merged**